### PR TITLE
Feature/gh 19 separate filtering from decisions

### LINKED ISSUE #19: Separate filtering from decision making
Library users often request that they'd like to retrieve decisions but no exception thrown. Filter class should be split into two.

### PR DESCRIPTION
**Pull request by lukasz-kaminski**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import java.util.Base64
 
 allprojects {
-    version = "0.0.5"
+    version = "1.0.0-SNAPSHOT"
```
Comment (irodzik) [inline build.gradle.kts:4]: Hold your horses! :horse_racing:

File: opa-filter-core/src/main/java/com/bisnode/opa/spring/security/filter/OpaFilter.java [hunk 2]
```diff
@@ -61,28 +53,18 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 
     private void decideFor(HttpServletRequest httpRequest) {
         try {
-            Decision decision = fetchDecision(httpRequest);
-            log.trace("OPA response is 'allow': {} for access to {} {}", decision.getAllow(), httpRequest.getMethod(), httpRequest.getServletPath());
-            if (!decision.isAllow()) {
-                denyAccess(decision);
+            AccessDecision accessDecision = decider.decideBy(httpRequest);
```
Comment (irodzik) [inline opa-filter-core/src/main/java/com/bisnode/opa/spring/security/filter/OpaFilter.java]: Once we have `decideFor` and then `decideBy`. I think we should unify it

**Review by irodzik — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (lukasz-kaminski) [reply build.gradle.kts:4]: Constructor of `OpaFilter` has changed. Whoever used it without a starter, now has a breaking change. According to SemVer, this has to be 1.0.0.

@anderseknert ?

**Review by irodzik — APPROVED** [review]

**Comment by irodzik** [comment]
Ok i got conviced that we should keep `1.0.0` version

**Review by anderseknert — APPROVED** [review]
Clean code and great tests, as always 👍 One thing I'm wondering about though is this approach vs. built in Spring Security stuff like the "voter" classes that seem to be used in the [contrib Spring examples](https://github.com/open-policy-agent/contrib/tree/master/spring_authz). Would it be a good option for this library to also easily allow for that approach?

**Comment by lukasz-kaminski** [comment]
> Clean code and great tests, as always  

Thanks!

> One thing I'm wondering about though is this approach vs. built in Spring Security stuff like the "voter" classes that seem to be used in the [contrib Spring examples](https://github.com/open-policy-agent/contrib/tree/master/spring_authz). Would it be a good option for this library to also easily allow for that approach?

I'm still not sure which way we should go. Both approaches are acceptable in terms of Spring Security. Voting has a huge drawback that you can't pass the reason for denial. Also, if you keep authorization policies decoupled (as you should when using OPA), I don't think there should be a case where OPA's decision can be outvoted.

Of course, we should let users do whatever they please with their code, so I'm OK with adding additional voter-based implementation and making it configurable (as opposed to the exception based, which would be autoconfigured by default).

I'd be grateful if you filed an issue if that's what you'd like to see.

**Pull request merged**